### DISCUSSION TITLE
👀 Watch routes changes and rebuild automatically

### DISCUSSION THREAD

**Discussion by sandulat** [sandulat/routes-gen#15] [Announcements]
It would be great if the `routes-gen` command accepted a `--watch` flag, to watch route files changes in background and rebuild the typings automatically.

**Comment by sandulat** [comment]
This is now being implemented in https://github.com/sandulat/routes-gen/releases/tag/v0.4.0

Upgrade both packages:
`yarn add routes-gen@latest @routes-gen/remix@latest`

You can now use the `--watch` or `-w` flag.

For Remix, `${appDirectory}/routes/**/*.{ts,tsx,js,jsx}` and `remix.config.js` are being watched for changes.

Any feedback would be much appreciated!

**Reply by aniravi24** [reply]
It's been working for us, thanks for this! One thing we do as soon as the file is created is to format/lint it because we've set up our tooling to reject code that doesn't meet our format/lint rules within our project. We're solving this problem by using npm-watch to listen for the file change and then run prettier, but one nice-to-have would be to have some kind of "post generate" function/hook to be able to run an arbitrary command after it's finished generating.

**Reply by sandulat** [reply]
@aniravi24 Sounds great. It's implemented and working. Example usage:
`routes-gen --post-export="npm run lint"`
`routes-gen --post-export="npm run lint" --watch`

I'll publish a new version these days, as soon as I have some free time. It needs some testing.

**Reply by sandulat** [reply]
PR: https://github.com/sandulat/routes-gen/pull/18

**Reply by sandulat** [reply]
@aniravi24 This is now released under v0.5.0:
https://github.com/sandulat/routes-gen/releases/tag/v0.5.0
https://www.npmjs.com/package/routes-gen/v/0.5.0

The usage is exactly the same as in the example above. Also documented in README.

Sorry for the delay!

**Reply by aniravi24** [reply]
No worries at all, excited to try it! Thanks so much!

**Reply by aniravi24** [reply]
This works great, thanks a bunch!

**Comment by robyhuzwandar** [comment]
hi @sandulat 
how does it actually work? when will it be automatically generated if I create a new route?
where should I put the command --watch ?

**Reply by sandulat** [reply]
Hey @robyhuzwandar, just append the flag right after the command you're running. `yarn routes-gen -d @routes-gen/remix --watch`

**Reply by robyhuzwandar** [reply]
Owh, that means we still need to run the script even if only once in every start to work right? @sandulat

**Reply by robyhuzwandar** [reply]
I got the issue that if I delete the folder immediately, let's say I have a folder:

`help/index.tsx` and I delete the `help` folder it won't auto update the routes.d.ts, but if I delete `index.tsx` it will update routes.d.tsx automatically
